persistence-jdbc/javadsl/src/test/scala/com/lightbend/lagom/javadsl/persistence/jdbc/JdbcPersistenceModuleSpec.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.javadsl.persistence.jdbc
+
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+import com.lightbend.lagom.internal.javadsl.api.broker.TopicFactory
+import com.lightbend.lagom.javadsl.api.ServiceLocator
+import com.typesafe.config.ConfigFactory
+import javax.inject.Inject
+import org.scalatest.{ BeforeAndAfterAll, Matchers, WordSpec, WordSpecLike }
+import play.api.Play
+import play.api.db.DBApi
+import play.api.inject.{ ApplicationLifecycle, DefaultApplicationLifecycle }
+import play.inject.guice.GuiceApplicationBuilder
+import play.api.inject.{
+  ApplicationLifecycle,
+  BindingKey,
+  DefaultApplicationLifecycle,
+  bind => sBind
+}
+
+/**
+ * This test will simply wire a minimal application using Guice.
+ *
+ * The LagomDbApiProvider is a copy from the Play that does not try to connect to the DB
+ * when created.
+ *
+ * In this particular test, we explicitly set a bogus DB configuration and let Guice create the class
+ * and wire then together. Because the Lagom impl does not connect eagerly no error should be thrown
+ * after wiring and the test should just pass.
+ */
+class JdbcPersistenceModuleSpec
+  extends WordSpecLike
+  with Matchers
+  with BeforeAndAfterAll {
+
+  "The JdbcPersistenceModule" should {
+    "wire Lagom's specific LagomDbApiProvider without throwing any error even in the absence of a properly configured DB" in {
+
+      val lifecycle = new DefaultApplicationLifecycle
+      val initialBuilder = new GuiceApplicationBuilder()
+        .overrides(sBind[ApplicationLifecycle].to(lifecycle))
+
+      // bogus DB configurations
+      val dbConfig =
+        """
+          |db.default {
+          |  driver = "org.bogus.Driver"
+          |  url = "jdbc:bogus://localhost:1234/bogus"
+          |  user = "bogus"
+          |  password = "bogus"
+          }
+        """.stripMargin
+
+      // this would blow up if we were eagerly creating DB connections
+      new GuiceApplicationBuilder()
+        .overrides(sBind[ApplicationLifecycle].to(lifecycle))
+        .bindings(sBind[DbWrapper].toSelf)
+        .configure(ConfigFactory.parseString(dbConfig))
+        .build()
```

Comment:
this is wrong, I know realised that the app have started in the background and we have to stop it. 

When stopping it, the test will fail because there is a shutdown hook to shutdown the db.
This will fail because of the bogus DB. 

I'm afraid that we can't make a test to prove that a bogus DB won't block the app to start.